### ISSUE TITLE
Hard to configure SpaceVim not to require external dependencies

### ISSUE THREAD

**Issue by cies** [wsdjeg/SpaceVim#3578]
I've used several vim starter kits in the past, and happily migrated to SpaceVim (after having some experience with Spacemacs). Currently I'm not interested to use SpaceVim as an IDE, I dont want it do all kinds of intelligent handling of my code (mostly by connecting to language backends) or to have it connect to my database when editing some SQL file. I want to simply want to use it as a text editor that works both from the CLI and using NeoVim, that provides enough fancy features with well-chosen defaults to keep me productive and happily surprised.

I explicitly do **not** want:

1. Annoying error messages about "Cannot connect to XYZ" (some language backend or SQL database)
2. Long file open times
3. Complex setup outside of adding some layers

Currently these 3 things do often happen when I just configure some layer for the few languages I commonly use.

SpaceVim is great, the layers are great, the community is great. Yet it does not work for me as a "casual" (non IDE) user that just want to flick a bunch of layers on and put a few personal settings in. I like SpaceVim default feature set (compared to the default vim settings on Linux box) when they work and work fast: pretty by default, self documenting space interface, community, layers for everything at my fingertips, plugin/dependency management, compatibility.
But I cannot easily switch off or otherwise configure my way out of the just mentioned annoyances just yet.

I tried asking about this on the Gitter chat and got some pointers, but they did not cut it.

This was kindly recommended:

```
[[layers]]
  name = 'checkers'
  enable = false
```

But it did not help much. Java files still loading dog slow, opening SQL files still made it try to connect to a database (seriously?!).

I believe SpaveVim would do better to default to no or *very few AND very well documented* external dependencies. And have setting (that are off by default) for layers to turn on such external dependencies. Possibly with some smart trick easily generating a list of all advanced features per layer one can unlock by installing external dependencies, and even info on how to do so on common platforms/ test to prove they were installed properly (after which one then can turn said advanced features on).

But I understand this may be too much to ask for. I get it: maybe I'm one of the few non-power users of SpaceVim. But I guess a little guide on how to get started with SpaceVim without seeing errors and slowness due to missing external dependencies would be reaaaally great.

My case is not to make SpaveVim not require any external deps at all. I would not mind if a "casual user" starter config would require a bunch of external deps. Just the need to have external deps for any layer I might flick on, which without it would yield errors and slowness is a grande nuisance.

Actually I hope some "casual user" config for SpaceVim exists, and one can show me, or we can find what needs to be in such config by suggestions in this issue. Thanks in advance for thinking with me.

**Comment by wsdjeg** [comment]
Sorry for late reply. I do not understand. do you mean disable checkers layer by default? spacevim do not load language layer by default. all default layers have been documented in spacevim.org/layers/

**Comment by cies** [comment]
I dont know what it is doing but I get errors with several languages (Java and SQL come to mind), because it is not able to find some external dependencies. Also starting vim while opening file from these languages is slow, and so is the "wq" command. I wonder if there is a way to slim SpaceVim down to not get so bloated.

My main point here is: SpaceVim is not fit for the use case of being a simple editor. It might be with some config, but I cannot figure out what that config is. Now what is a simple editor you may ask? I'd say: 1) does not require additional programs to be installed, 2) does not provide a full IDE experience, 3) starts and stops very quickly on any file.

**Comment by wsdjeg** [comment]
@cies without Exact reproduction steps, I can not do anything to fix this issue.

1. does not require additional programs to be installed.

By default, spacevim does not require  additionall programs. If you get error about missing programs, please post the error message.

2. does not provide a full IDE experience

what is full IDE experience? which feature should be disabled by default? you should point to the feature.

3. starts and stops quickly

if you need quick start, you should use `vim -u NONE`, it is fastest.

**Comment by Rom1deTroyes** [comment]
> if you need quick start, you should use `vim -u NONE`, it is fastest.

`vim --noplugin` is a little less aggressive in the way ;-)

Having a special vimrc for special usecase can be an other aproche : `vim -u ~/.vim-python.rc`

**Comment by cies** [comment]
I've just installed a clean SpaceVim + NeoVim combo and configured it to use a bunch of layers.

The hardest bit was installing `vimproc.vim`, I used these commands for that:

```
cd /usr/local/src/
git clone https://github.com/Shougo/vimproc.vim; cd vimproc.vim/
cp -r autoload/vimproc* ~/.config/nvim/autoload/
cp -r plugin ~/.config/nvim/
cp -r lib ~/.config/nvim/
ln -s /usr/bin/nvim /usr/local/bin/vim
ln -s /usr/bin/nvim /usr/local/bin/vi
```

Starting NeoVim with the layer installed produced some more errors (all "Could not build dependency" stuff) which I simply ignored.

I was happily surprised to find all my issues where fixed: no more slow load for big Java files, no more errors when merely opening Java and SQL files, ignoring the build errors in my layers did not cause any problems so far.

Thanks @wsdjeg  for taking it seriously. I leave closing the issue up to you. Just know I'm happy with the way things are now.
